app/Services/UserRoleService.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\UserRoles;
+
+class UserRoleService
+{
+    public static function can($action, $id)
+    {
+        $roles = self::get($id);
+
+        return in_array($action, $roles) ? $roles[$action] : null;
+    }
+
+    public static function get($id)
+    {
+        if($roles = UserRoles::whereUserId($id)->first()) {
+            return $roles->roles;
+        }
+
+        return self::defaultRoles();
+    }
+
+    public static function roleKeys()
+    {
+        return array_keys(self::defaultRoles());
+    }
+
+    public static function defaultRoles()
```

Comment:
Following the comment around having a fixed set of roles, this would become your "default role" assigned to users (in Mastodon it has the `role_id` as nullable, and infers the default this way, as there's a "-99" default role created: https://github.com/mastodon/mastodon/blob/46bd58f74d11591a0180319285b0c79b2212ef69/app/models/user_role.rb#L105 )

Reply:
I just pushed an updated defaultRoles!